src/geom.hpp

```diff
@@ -117,6 +120,19 @@ class linestring_t : public point_list_t
         return 1;
     }
 
+    /**
+     * Check validity of linestring.
+     *
+     * By the OGC definition a linestring must have at least two distinct
+     * nodes to be valid. We are a bit more strict and require that the
+     * first two nodes are also distinct. When we create linestrings from
+     * OSM data this will always be the case.
+     */
+    bool is_valid() const noexcept
+    {
+        return size() > 1 && (*this)[0] != (*this)[1];
```

Comment:
Is the second check still needed now that you are explicitly removing duplicates?

Reply:
Not in the case we have so far, but I would expect that this function will be useful for other cases where we didn't explicitly remove duplicates, so I'd rather keep that.

Reply:
I have remove the is_valid() function now and do the check without it. This should be less confusing.